EKF/SensorRangeFinder.hpp

```diff
@@ -0,0 +1,131 @@
+/****************************************************************************
+ *
+ *   Copyright (c) 2020 Estimation and Control Library (ECL). All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ * 3. Neither the name ECL nor the names of its contributors may be
+ *    used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+/**
+ * @file SensorRangeFinder.hpp
+ * Range finder class containing all the required checks
+ *
+ * @author Mathieu Bresciani <brescianimathieu@gmail.com>
+ *
+ */
+#pragma once
+
+#include "Sensor.hpp"
+#include <matrix/math.hpp>
+
+using estimator::rangeSample;
+using matrix::Dcmf;
+
+class SensorRangeFinder : public Sensor
+{
+public:
+	SensorRangeFinder() = default;
+	~SensorRangeFinder() override = default;
+
+	void runChecks(uint64_t time_delayed_us, const Dcmf &R_to_earth);
+	bool isHealthy() const override { return _rng_hgt_valid; }
+	bool isNewHealthyData() const override { return _range_data_ready && _rng_hgt_valid; }
+	bool isDelayedHealthyData() const override { return _range_data_ready && _rng_hgt_valid; }
+	bool canBeusedAsFailover() const override;
+	bool canResetOnSensor() const override;
+
+	void setNewestSample(rangeSample sample) { _newest_sample = sample; }
```

Comment:
`setNewestSample() and `_newest_sample` is not used at the moment. I understand that the interface can then be extended to other sensors easily. But should we also allocate memory for `_newest_sample` and allow setting it, even if it is not used?